tests/jerry/es2015/regexp-prototype-match.js

```diff
@@ -0,0 +1,56 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+var regexp = /[0-9]+/g;
+var str = '2016-01-02';
+var num = 12131233;
+
+// Test with string input
+var result = regexp[Symbol.match](str);
```

Comment:
Is this functionality only accessible via the `Symbol.match`?

Reply:
In addition this functionality means that if someone creates a "subclass" of RegExp where the `Symbol.match` or `exec`(?) is implemented the "new" method would be invoked instead of the `Regexp.prototype.exec`? If so would it be possible to have a test case for it?